packages/api/controllers/nomination.js

```diff
@@ -137,15 +140,26 @@ const updateNomination = async (req, res) => {
 
       if (nomination.status === NOMINATION_STATUS.awaiting) {
         try {
-          nomination.update({ awaitingHipaaTimestamp: Date() });
+          await nomination.update({ awaitingHipaaTimestamp: Date() });
           const lastName = nomination.patientName
             ? nomination.patientName.split(' ')[1]
             : '';
           const state = states.getStateCodeByStateName(
             nomination.hospitalState
           );
+
           const applicationName = `${lastName}-${state}`;
-          createFolder(applicationName);
+          let driveFolderId = await createFolder(applicationName, nomination);
+
+          await nomination.update({
+            driveFolderId: driveFolderId,
+          });
```

Comment:
I'd probably combine the `awaitingHipaaTimestamp` update and the `driveFolderId` update into one call so we only hit the database twice (or three times, with the fetch call below). It also means there's less likelihood of a failure (say in the google drive API) causing the nomination to get into a bad state.

Strictly speaking, I'd probably roll the update call on 119 into that single call, too, but that'd require some substantial restructuring, and we can probably leave it.

Reply:
fixed